code/modules/mob/mob_helpers.dm

```diff
@@ -516,7 +516,11 @@ mob/proc/get_standard_bodytemperature()
 
 	var/obj/item/active_item = get_active_held_item()
 	if((istype(active_item) && active_item.unique_action(src) != COMSIG_KB_NOT_ACTIVATED) || client?.prefs.unique_action_use_active_hand)
+		if(active_item != null)
+			SEND_SIGNAL(active_item, COMSIG_ITEM_UNIQUE_ACTION, src)
```

Comment:
nop, unique_action can be called by different means (verb and click). 

Since you prefer to keep unique action as the actionner, we can use the old solution for the signal sent.

every time something is calling unique_action, have a wrapper proc that sends the signal and then call unique action/

This is awfull of course, but the proper fix would require a heavy refactor and gun and that's shit

Reply:
I wanted to keep it as close to original as possible, or if not possible, at least convenient to use. Right-clicking the weapon sprite wasn't convenient IMO. Other solution I can propose is to make reagents autoload into the blade.